src/histolab/slide.py

```diff
@@ -135,6 +145,59 @@ def level_dimensions(self, level: int = 0) -> Tuple[int, int]:
                 f"{len(self._wsi.level_dimensions)}"
             )
 
+    def level_magnification_factor(self, level: int = 0) -> str:
+        """Return the magnification factor at the specified level.
+
+         Notice that the conversion level-magnification can be computed only
+        if the native magnification is available in the slide metadata.
+
+        Parameters
+        ---------
+        level : int
+            The level which magnification factor is requested, default is 0.
+
+        Returns
+        -------
+        magnification factor : str
+            Magnification factor at speficied level
+
+        Raises
+        ------
+        LevelError
+            If the specified level is not available.
+        SlidePropertyError
+            If the slide's native magnification is not available in the file
+            metadata.
+        """
+        level = level if level >= 0 else self._remap_level(level)
+        properties = self.properties
+
+        if level not in self.levels:
+            raise LevelError(
+                f"Level {level} not available. Number of available levels: "
+                f"{len(self._wsi.level_dimensions)}"
```

Comment:
here you should also use `len(self.levels)`, and then you will see that the unit test `and_it_raises_an_exception_if_level_in_incorrect` will fail (as it should)